### ISSUE TITLE
Dataset: unknown options should warn

### ISSUE THREAD

**Issue by ax3l** [openPMD/openPMD-api#1095]
I realized that if we use unknown options in `Dataset(..., options=...)` that this does not warn.

We had a typo in WarpX for the "resizable" key that was silently ignored and then threw later on in resizing from the HDF5 layer.

**Comment by franzpoeschel** [comment]
This is generally something that we already do have:
```
> bin/SerialIOTests serial_adios2_json_config
Filters: serial_adios2_json_config
Warning: ADIOS2 backend does not support compression method blosc. Continuing without compression.
Warning: parts of the JSON configuration for ADIOS2 remain unused:
{"engine":{"unused":"parameter"},"unused":"as well"}
Warning: ADIOS2 backend does not support compression method blosc. Continuing without compression.
Warning: parts of the JSON configuration for ADIOS2 dataset '/data/0/meshes/E/y' remain unused:
{"dataset":{"unused":"too"},"unused":"dataset parameter"}
Warning: ADIOS2 backend does not support compression method blosc. Continuing without compression.
Warning: parts of the JSON configuration for ADIOS2 remain unused:
{"engine":{"unused":"parameter"},"unused":"as well"}
Warning: ADIOS2 backend does not support compression method blosc. Continuing without compression.
Warning: parts of the JSON configuration for ADIOS2 dataset '/data/0/meshes/E/y' remain unused:
{"dataset":{"unused":"too"},"unused":"dataset parameter"}
Warning: ADIOS2 backend does not support compression method blosc. Continuing without compression.
Warning: parts of the JSON configuration for ADIOS2 remain unused:
{"engine":{"unused":"parameter"},"unused":"as well"}
Warning: ADIOS2 backend does not support compression method blosc. Continuing without compression.
Warning: parts of the JSON configuration for ADIOS2 dataset '/data/0/meshes/E/y' remain unused:
{"dataset":{"unused":"too"},"unused":"dataset parameter"}
Warning: parts of the JSON configuration for ADIOS2 remain unused:
{"engine":{"unused":"parameter"}}
Warning: parts of the JSON configuration for ADIOS2 remain unused:
{"engine":{"unused":"parameter"}}
```

Since there are several places in our code base that parse JSON, it's entirely possible that this functionality is forgotten here or there. This being said, for me the warning does work in the following example:

```c++
{
    Series series( "temp.h5", Access::CREATE );
    auto E_x = series.iterations[ 0 ].meshes[ "E" ][ "x" ];
    Dataset ds( Datatype::INT, { 1 } );
    std::string jsonConfig = R"(
{
    "hdf5": {
        "dataset": {
            "resizable": true
        }
    }
})";
    ds.options = jsonConfig;
    E_x.resetDataset( ds );
    E_x.storeChunk( std::make_shared< int >( 8 ), { 0 }, { 1 } );
    E_x.seriesFlush();
}
```

Creates the following output:

```
Warning: parts of the JSON configuration for HDF5 dataset 'x' remain unused:
{"resizable":true}
```

Are you sure there was not another issue involving this? Can you give me a way to reproduce this?

**Comment by ax3l** [comment]
Thank you for the reproducer. The problem seems to be with this syntax that I somehow did not properly construct the shadow to warn:
https://github.com/openPMD/openPMD-api/blob/0.14.2/examples/3b_write_resizable_particles.cpp#L48

Code:
```C++
{
    Series series( "temp.h5", Access::CREATE );
    auto E_x = series.iterations[ 0 ].meshes[ "E" ][ "x" ];
    Dataset ds( Datatype::INT, { 1 } );
    std::string jsonConfig = "{ \"resizaAAAaaableeaux\": true }";
    ds.options = jsonConfig;
    E_x.resetDataset( ds );
    E_x.storeChunk( std::make_shared< int >( 8 ), { 0 }, { 1 } );
    E_x.seriesFlush();
}
```

**Comment by franzpoeschel** [comment]
This is now fixed with #1043: https://github.com/openPMD/openPMD-api/pull/1043#issuecomment-915358566
